### PR TITLE
Implement the necessary aeson instances for pool-params query for Alonzo

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs [hunk 1]
```diff
@@ -796,6 +802,11 @@ obtainLedgerEraClassConstraints
...
-obtainLedgerEraClassConstraints ShelleyBasedEraAlonzo  _ =
-  panic "obtainLedgerEraClassConstraints: Alonzo era not implemented yet"
-
+obtainLedgerEraClassConstraints ShelleyBasedEraAlonzo  f = f
+
+instance ToJSON (Alonzo.TxOut (Alonzo.AlonzoEra StandardCrypto)) where
+  toJSON (Alonzo.TxOut addr value dataHash) = Aeson.object
+    [ "addr" .= toJSON addr
+    , "value" .= toJSON value
+    , "dataHash" .= toJSON dataHash
```
Comment (newhoggy) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs:811]: `TxOut` field names.

File: cardano-api/src/Cardano/Api/Orphans.hs [hunk 2]
```diff
@@ -369,3 +372,65 @@ instance ToJSON Alonzo.AlonzoGenesis where
       , "collateralPercentage" .= Alonzo.collateralPercentage v
       , "maxCollateralInputs" .= Alonzo.maxCollateralInputs v
       ]
+
+instance ToJSON (Alonzo.PParams era) where
```
Comment (erikd) [inline cardano-api/src/Cardano/Api/Orphans.hs:376]: I really hate orphan instances, but at least these are in an `Orphans` module.

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs (hunk 1 above)
Comment (erikd) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs:811]: Sorry, what is that comment about?

**Review by erikd — APPROVED** [review]
LGTM!

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs:811]: I added to comment so I can make a permalink to share with the ledger team.

**Comment by newhoggy** [comment]
bors merge

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs:808]: We already have this instance in `Cardano.Tracing.OrphanInstances.Shelley`. What we should do is move the instance in `Cardano.Tracing.OrphanInstances.Shelley` to `Cardano.Api.Orphans` and import it instead.

File: cardano-api/src/Cardano/Api/Orphans.hs [hunk 3]
```diff
@@ -369,4 +375,76 @@ instance ToJSON Alonzo.AlonzoGenesis where
...
+        <*> obj .: "prices"
+        <*> obj .: "maxTxExUnits"
+        <*> obj .: "maxBlockExUnits"
+        <*> obj .: "maxValSize"
+        <*> obj .: "collateralPercentage"
+        <*> obj .: "maxCollateralInputs"
+
+deriving instance ToJSON (Alonzo.PParamsUpdate (Alonzo.AlonzoEra StandardCrypto))
+
+instance (Ledger.Era era, Show (Ledger.Value era), ToJSON (Ledger.Value era))
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Orphans.hs:440]: :100:

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**